modules/apps/frontend-js/frontend-js-spa-web/src/main/java/com/liferay/frontend/js/spa/web/internal/servlet/taglib/helper/SPAHelper.java

```diff
@@ -232,9 +239,32 @@ private long _getCacheExpirationTime(SPAConfiguration spaConfiguration) {
 		return cacheExpirationTime;
 	}
 
-	private static final String _REDIRECT_PARAM_NAME;
+	private JSONArray _getExcludedPathsJSONArray(
+		SPAConfiguration spaConfiguration) {
+
+		JSONArray jsonArray = JSONFactoryUtil.createJSONArray();
+
+		String[] excludedPaths = StringUtil.split(
+			SPAConfigurationUtil.get("spa.excluded.paths"));
 
-	private static final JSONArray _SPA_EXCLUDED_PATHS_JSON_ARRAY;
+		for (String excludedPath : excludedPaths) {
+			jsonArray.put(_portal.getPathContext() + excludedPath);
+		}
+
+		String[] customExcludedPaths = spaConfiguration.customExcludedPaths();
+
+		if (customExcludedPaths == null) {
+			return jsonArray;
+		}
```

Comment:
Very far from being proficient with Java, so just a thought - do we want to `return` sooner to save up some resources? Depends in what state we want `jsonArray` to be at the point of `return`, so I might be wrong here.

Reply:
This is a common pattern in the portal, and yes, we return it here to save resources and horizontal space in the code.